### PR TITLE
Fix the finalizer code

### LINKED ISSUE #1260: handle AddFinalizer properly
/kind bug
/area provider/ibmcloud

**What steps did you take and what happened:**
https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/16127b18afd2e019d75861ade36f3047f7924a67/controllers/ibmvpccluster_controller.go#L112

https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/16127b18afd2e019d75861ade36f3047f7924a67/controllers/ibmvpccluster_controller.go#L112

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by mkumatag**
**What this PR does / why we need it**:

This PR fixes following things:
- `controllerutil.AddFinalizer` already contains the logic for checking existing finilizer, hence the explicit check is removed
- Forcing the reconcile when the Finalizer gets add so that we don't leave behind any associated resources when delete triggers before the finalizer get added.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1260 

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by Karthik-K-N** [comment]
Changes looks good to me, We need to fix the mock calls.

**Comment by mkumatag** [comment]
> Changes looks good to me, We need to fix the mock calls.

I was anticipating this :) Lemme fix them

**Comment by mkumatag** [comment]
> Changes looks good to me, We need to fix the mock calls.

Fixed the unit tests, ptal  at the latest commit.

File: controllers/ibmpowervsmachine_controller_test.go [hunk 1]
```diff
@@ -447,7 +447,8 @@ func TestIBMPowerVSMachineReconciler_ReconcileOperations(t *testing.T) {
 				},
 				IBMPowerVSMachine: &infrav1beta2.IBMPowerVSMachine{
 					ObjectMeta: metav1.ObjectMeta{
-						Name: *pointer.String("capi-test-machine"),
+						Name:       *pointer.String("capi-test-machine"),
```
Comment (Karthik-K-N) [inline controllers/ibmpowervsmachine_controller_test.go:450]: Not related to this PR, may be later this assignment can be simplified.

**Pull request merged**